integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -1571,6 +1624,198 @@ describe.skipIf(environment.missingServer, "Flexible sync", function () {
         });
       });
 
+      describe("Results#subscribe", function () {
+        it("waits for objects to sync the first time only", async function (this: RealmContext) {
+          expect(this.realm.subscriptions).to.have.length(0);
+
+          const peopleOver10 = this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10");
+
+          // Subscribing the first time should wait for synchronization.
+          await peopleOver10.subscribe({ behavior: WaitForSync.FirstTime });
+          expect(this.realm.subscriptions.state).to.equal(SubscriptionSetState.Complete);
+
+          // Subscribing the second time should return without waiting.
+          await peopleOver10.subscribe({ behavior: WaitForSync.FirstTime });
+          expect(this.realm.subscriptions.state).to.equal(SubscriptionSetState.Pending);
+
+          expect(this.realm.subscriptions).to.have.length(1);
+        });
+
+        it("waits for objects to sync the first time only by default", async function (this: RealmContext) {
+          expect(this.realm.subscriptions).to.have.length(0);
+
+          const peopleOver10 = this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10");
+
+          await peopleOver10.subscribe();
+          expect(this.realm.subscriptions.state).to.equal(SubscriptionSetState.Complete);
+
+          await peopleOver10.subscribe();
+          expect(this.realm.subscriptions.state).to.equal(SubscriptionSetState.Pending);
+
+          expect(this.realm.subscriptions).to.have.length(1);
+        });
+
+        it("waits for objects to sync the first time only for different 'Results' instances", async function (this: RealmContext) {
```

Comment:
Maybe the test title is a little confusing for me, but this means that `WaitForSync` applies to the query and not to the results, am I correct?

Reply:
This is due to how we determine if we should wait for sync or not. If, for instance, the subscription is not named, we check if it already exists *by query* like you said.

I'll modify the test description and can also add some comments in the test 👍